File: tests/rspec/pages/installer/_install_form.rb
module Installer
  class Form < SitePrism::Section
    element :db_hostname, 'input[name=db_hostname]'
    element :db_name, 'input[name=db_name]'
    element :db_username, 'input[name=db_username]'
    element :db_password, 'input[name=db_password]'
    element :db_prefix, 'input[name=db_prefix]'

    # TODO: Enable this when we add the default theme back
    # elements :install_default_theme, 'input[name=install_default_theme]'

    element :username, 'input[name=username]'
    element :email_address, 'input[name=email_address]'
    element :password, 'input[name=password]'

    element :install_submit, 'form input[type=submit]'
  end
end

File: tests/cypress/cypress/elements/pages/installer/_install_form.js
import ControlPanel from '../ControlPanel'

class Form extends ControlPanel {
    constructor() {
        super()

		this.all_there = {
			"db_hostname": 'input[type!=hidden][name=db_hostname]',
			"db_name": 'input[type!=hidden][name=db_name]',
			"db_username" : 'input[type!=hidden][name=db_username]',
			"db_password" : 'input[type!=hidden][name=db_password]',
			"db_prefix" : 'input[type!=hidden][name=db_prefix]',
			"install_default_theme" : 'input[type!=hidden][name=install_default_theme]',
			"username" : 'input[type!=hidden][name=username]',
			"email_address" : 'input[type!=hidden][name=email_address]',
			"password" : 'input[type!=hidden][name=password]',
			"license_agreement" : 'input[type!=hidden][name=license_agreement]',
			"install_submit" : 'form input[type=submit]'
		}
	    this.selectors = Object.assign(this.selectors, this.all_there)

	}
}
export default Form;